storagemarket/types.go

```diff
@@ -192,3 +194,26 @@ type ProviderDealState struct {
 	DealID        abi.DealID
 	FastRetrieval bool
 }
+
+// ListDealsPageParams configures the params which will be used to paginate & filter the deals that are returned by the
+// list deals commands.
+type ListDealsPageParams struct {
+	// CreationTimePageOffset is the minimum creation time of the deals which will be included in the deal list page.
+	CreationTimePageOffset time.Time
+
+	// DealsPerPage configures the number of deals to show on the deal list page.
+	// Defaults to math.MaxInt32.
+	DealsPerPage int
+
+	// MinStartEpoch configures the minimum start epoch of the deals which will be included in the deal list page.
+	// If this field is not configured, it becomes a no-op.
+	MinStartEpoch abi.ChainEpoch
+
+	// MaxEndEpoch configures the maximum end epoch of the deals which will be included in the deal list page.
+	// If this field is not configured, it becomes a no-op.
+	MaxEndEpoch abi.ChainEpoch
+
+	// ShowStorageDealError will also include deals which are in the `ShowStorageDealError` state.
+	// Defaults to false.
+	ShowStorageDealError bool
```

Comment:
I'd suggest we return all deals by default.
So let's call this `HideDealsInErrorState`

Reply:
@dirkmc That's now what Lotus does currently though. Do you still wanna do it this way ?

Reply:
I think so, yeah - it seems more intuitive to me to return everything by default, and then apply filters as requested.

We can just update the lotus code to match this interface instead of the other way around.

Reply:
This is done.